pkg/virt-controller/watch/snapshot.go

```diff
@@ -684,28 +686,46 @@ func (s *vmSnapshotSource) Lock() (bool, error) {
 
 	log.Log.Infof("Adding VM snapshot finalizer to %s", s.vm.Name)
 
+	var err error
 	vmCopy := s.vm.DeepCopy()
-	vmCopy.Status.SnapshotInProgress = &s.snapshot.Name
-	controller.AddFinalizer(vmCopy, sourceFinalizer)
 
-	_, err := s.client.VirtualMachine(vmCopy.Namespace).UpdateStatus(vmCopy)
-	if err != nil {
-		return false, err
+	if vmCopy.Status.SnapshotInProgress == nil {
+		vmCopy.Status.SnapshotInProgress = &s.snapshot.Name
+		vmCopy, err = s.client.VirtualMachine(vmCopy.Namespace).UpdateStatus(vmCopy)
```

Comment:
We released the VM resource without the status subresource enabled. You will have to use the status updater helper for it. Snapshots are not yet released, there it wlill be good enough to just us `Update` and UpdateStatus`. The helpers are here: https://github.com/kubevirt/kubevirt/blob/master/pkg/util/status/status.go#L242. This updater can on the fly react to enabling the status subresource. Best to make it a member of the controller struct itself, since it has internal caching to avoid unnecessary probing.